### PR TITLE
Add database merge tool

### PR DESCRIPTION
**Pull request by muXxer**
This PR adds a tool that can be used to merge two databases.

### CODE REVIEW

File: pkg/toolset/database.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	CfgProtocolPublicKeyRangesJSON = "publicKeyRanges"
+	// subfolder for the tangle database
+	TangleDatabaseDirectoryName = "tangle"
+	// subfolder for the UTXO database
+	UTXODatabaseDirectoryName = "utxo"
+)
+
+func getMilestoneManagerFromConfigFile(filePath string) (*milestonemanager.MilestoneManager, error) {
+
+	nodeConfig, err := loadConfigFile("config.json")
```
Comment (alexsporn) [inline pkg/toolset/database.go]: why is this hardcoded?

File: core/database/split.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func SplitIntoTangleAndUTXO(databasePath string) error {
 	utxoDatabasePath := filepath.Join(legacyDatabasePath, UTXODatabaseDirectoryName)
 
 	// Read the engine the current database is using
-	dbEngine, err := database.CheckDatabaseEngine(legacyDatabasePath, false)
+	dbEngineLegacy, err := database.CheckDatabaseEngine(legacyDatabasePath, false, dbEngine...)
```
Comment (alexsporn) [inline core/database/split.go]: don't like the `legacy` in the name

File: pkg/toolset/database.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+const (
+	// the amount of public keys in a milestone.
+	CfgProtocolMilestonePublicKeyCount = "protocol.milestonePublicKeyCount"
+	// the ed25519 public key of the coordinator in hex representation.
+	CfgProtocolPublicKeyRanges = "protocol.publicKeyRanges"
+	// the ed25519 public key of the coordinator in hex representation.
+	CfgProtocolPublicKeyRangesJSON = "publicKeyRanges"
+	// subfolder for the tangle database
+	TangleDatabaseDirectoryName = "tangle"
```
Comment (alexsporn) [inline pkg/toolset/database.go]: there are constants for these already

File: pkg/toolset/database.go (hunk 1 above)
Comment (muXxer) [reply pkg/toolset/database.go]: lol... that slipped through :sweat_smile:

File: pkg/toolset/database.go (hunk 3 above)
Comment (muXxer) [reply pkg/toolset/database.go]: I didn't want to import the "core plugins" here :thinking:

Comment (muXxer) [reply pkg/toolset/database.go]: Ah lel, I already imported them anyway.

**Review by alexsporn — APPROVED** [review]

**Review by alexsporn — APPROVED** [review]

**Pull request merged**